php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -50,6 +75,157 @@ public ExpressionTree EXPRESSION() {
+          ENCAPSULATED_COMPLEX_VARIABLE()
+        )
+      );
+  }
+
+  public ExpressionTree ENCAPSULATED_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedComplexVariable(
+        b.token(LCURLYBRACE),
+        b.token(PHPLexicalGrammar.NEXT_IS_DOLLAR),
+        EXPRESSION(),
+        b.token(RCURLYBRACE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_SEMI_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SEMI_COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedSemiComplexVariable(
+        b.token(DOLLAR_LCURLY),
+        b.firstOf(
+          EXPRESSION(),
+          f.expressionRecovery(b.token(PHPLexicalGrammar.SEMI_COMPLEX_RECOVERY_EXPRESSION))),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_SIMPLE_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SIMPLE_ENCAPS_VARIABLE)
+    .is(
+      f.encapsulatedSimpleVar(ENCAPSULATED_VARIABLE_IDENTIFIER(),
+        b.optional(b.firstOf(
+          f.expandableArrayAccess(
+              b.token(LBRACKET),
+            b.firstOf(IDENTIFIER(), NUMERIC_LITERAL(), ENCAPSULATED_VARIABLE_IDENTIFIER()),
+              b.token(RBRACKET)),
+            f.expandableObjectMemberAccess(b.token(ARROW), IDENTIFIER()))))
+    );
+  }
+
+  public IdentifierTree IDENTIFIER() {
+    return b.<IdentifierTree>nonterminal(Kind.IDENTIFIER)
+      .is(f.identifier(b.token(PHPLexicalGrammar.IDENTIFIER)));
+  }
+
+  public VariableIdentifierTree ENCAPSULATED_VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR_IDENTIFIER)
+      // variable identifiers encapsulated into strings literal does not allowed line terminator spacing,
+      // so here using "WHITESPACE" instead of SPACING.
+      .is(f.encapsulatedVariableIdentifier(b.token(PHPLexicalGrammar.WHITESPACES), b.token(PHPLexicalGrammar.VARIABLE_IDENTIFIER)));
+  }
+
+  public ExpressionTree EXPANDABLE_STRING_CHARACTERS() {
+    return b.<ExpandableStringCharactersTree>nonterminal(Kind.EXPANDABLE_STRING_CHARACTERS)
+      .is(f.expandableStringCharacters(b.token(PHPLexicalGrammar.STRING_WITH_ENCAPS_VAR_CHARACTERS)));
+  }
+
+  public YieldExpressionTree YIELD_EXPRESSION() {
+    return b.<YieldExpressionTree>nonterminal(Kind.YIELD_EXPRESSION)
+      .is(f.yieldExpression(
+        b.token(YIELD),
+        EXPRESSION(),
+        b.optional(f.newTuple1(b.token(DOUBLEARROW), EXPRESSION()))
+      ));
+  }
+
+  public ParenthesisedExpressionTree PARENTHESIZED_EXPRESSION() {
+    return b.<ParenthesisedExpressionTree>nonterminal(Kind.PARENTHESISED_EXPRESSION)
+      .is(f.parenthesizedExpression(
+        b.token(LPARENTHESIS),
+        b.firstOf(
+          YIELD_EXPRESSION(),
+          EXPRESSION()),
+        b.token(RPARENTHESIS)
+      ));
+  }
+
+  public AssignmentExpressionTree LIST_EXPRESSION_ASSIGNMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)